test/e2e/util/util.go

```diff
@@ -587,13 +594,39 @@ func EnsureNoCrashingPods(t *testing.T, ctx context.Context, client crclient.Cli
 			}
 			for _, containerStatus := range pod.Status.ContainerStatuses {
 				if containerStatus.RestartCount > crashToleration {
+					if isLeaderElectionFailure(ctx, guestClient, &pod, containerStatus.Name) {
+						t.Logf("Leader election failure detected in container %s in pod %s", containerStatus.Name, pod.Name)
+						continue
+					}
 					t.Errorf("Container %s in pod %s has a restartCount > 0 (%d)", containerStatus.Name, pod.Name, containerStatus.RestartCount)
 				}
 			}
 		}
 	})
 }
 
+func isLeaderElectionFailure(ctx context.Context, guestClient *kubeclient.Clientset, pod *corev1.Pod, containerName string) bool {
```

Comment:
Maybe export the function and add a comment so that it's reusable and discoverable for future tests? no strong opinion though